AceComm-3.0/ChatThrottleLib.lua

```diff
@@ -353,40 +427,60 @@ function ChatThrottleLib.OnUpdate(this,delay)
 		return -- argh. some bastard is spewing stuff past the lib. just bail early to save cpu.
 	end
 
-	-- See how many of our priorities have queued messages (we only have 3, don't worry about the loop)
-	local n = 0
-	for prioname,Prio in pairs(self.Prio) do
-		if Prio.Ring.pos or Prio.avail < 0 then
-			n = n + 1
+	-- Integrate blocked queues back into their rings periodically.
+	if self.BlockedQueuesDelay >= 1 then
```

Comment:
I do not see any gain in artificial delay added to readding blocked ones to ring. Putting queues at the end of `Ring` would be enough (as moving through ring should give you some delay) and would allow you to drop the `Blocked` ring completely.

One concern that I do not think I understand is what would happen if all queues are blocked at the same time?

Reply:
This was also discussed in the comments and is still pending resolution.